bug_update.php

```diff
@@ -63,401 +63,187 @@
-) {
-	if( $t_updated_bug->handler_id != NO_USER ) {
-		$t_updated_bug->status = config_get( 'bug_assigned_status' );
-	} else {
-		$t_updated_bug->status = config_get( 'bug_submit_status' );
-	}
+$t_target_version = gpc_get_string( 'target_version', '' );
+if( !is_blank( $t_target_version ) ) {
+    $t_issue['target_version'] = array( 'name' => $t_target_version );
 }
 
-# Handle automatic assignment of issues.
-$t_updated_bug->status = bug_get_status_for_assign( $t_existing_bug->handler_id, $t_updated_bug->handler_id, $t_existing_bug->status, $t_updated_bug->status );
+$t_version = gpc_get_string( 'product_version', '' );
+if( !is_blank( $t_version ) ) {
+    $t_issue['version'] = array( 'name' => $t_version );
+}
 
-# Allow a custom function to validate the proposed bug updates. Note that
-# custom functions are being deprecated in MantisBT. You should migrate to
-# the new plugin system instead.
-helper_call_custom_function( 'issue_update_validate', array( $f_bug_id, $t_updated_bug, $t_bug_note->note ) );
+$t_view_state = gpc_get_int( 'view_state', 0 );
+if( $t_view_state != 0 ) {
+    $t_issue['view_state'] = array( 'id' => $t_view_state );
+}
 
-# Allow plugins to validate/modify the update prior to it being committed.
-$t_updated_bug = event_signal( 'EVENT_UPDATE_BUG_DATA', $t_updated_bug, $t_existing_bug );
+$t_bugnote = array();
 
-# Commit the bug updates to the database.
-$t_text_field_update_required = ( $t_existing_bug->description != $t_updated_bug->description ) ||
-								( $t_existing_bug->additional_information != $t_updated_bug->additional_information ) ||
-								( $t_existing_bug->steps_to_reproduce != $t_updated_bug->steps_to_reproduce );
-$t_updated_bug->update( $t_text_field_update_required, true );
+$t_bugnote_text = gpc_get_string( 'bugnote_text', '' );
+if( !is_blank( $t_bugnote_text ) ) {
+    $t_bugnote['bugnote_text'] = $t_bugnote_text;
+}
 
-# Update custom field values.
-foreach ( $t_custom_fields_to_set as $t_custom_field_to_set ) {
-	custom_field_set_value( $t_custom_field_to_set['id'], $f_bug_id, $t_custom_field_to_set['value'] );
+$t_private = gpc_get_bool( 'private' );
+if( $t_private != 0 ) {
+    $t_bugnote['private'] = VS_PRIVATE;
 }
 
-# Add a bug note if there is one.
-if( $t_bug_note->note || helper_duration_to_minutes( $t_bug_note->time_tracking ) > 0 ) {
-	$t_bugnote_id = bugnote_add( $f_bug_id, $t_bug_note->note, $t_bug_note->time_tracking, $t_bug_note->view_state == VS_PRIVATE, 0, '', null, false );
-	bugnote_process_mentions( $f_bug_id, $t_bugnote_id, $t_bug_note->note );
+$t_time_tracking = gpc_get_string( 'time_tracking', '' );
+if( !is_blank( $t_time_tracking ) ) {
+    $t_bugnote['time_tracking'] = $t_time_tracking;
 }
 
-# Add a duplicate relationship if requested.
-if( $t_updated_bug->duplicate_id != 0 ) {
-	relationship_upsert( $f_bug_id, $t_updated_bug->duplicate_id, BUG_DUPLICATE, /* email_for_source */ false );
+# Validate the custom fields before adding the bug.
```

Comment:
@vboctor 
I borrowed the custom fields from bug_report.php.
How can I get all the "custom fields" using the gpc_get_custom_field function?